homeassistant/helpers/service.py

```diff
@@ -112,3 +116,46 @@ def extract_entity_ids(hass, service_call, expand_group=True):
             return [service_ent_id]
 
         return service_ent_id
+
+
+@bind_hass
+def get_description(hass, domain, service):
+    """Return the description (i.e. user documentation) for a service call."""
+    if domain == ha.DOMAIN:
+        import homeassistant.components as components
+        comp_file = components.__file__
+    else:
+        comp_file = get_component(domain).__file__
+
+    descriptions = DESCRIPTION_CACHE.get(comp_file)
+    if descriptions is None:
+        descriptions = load_yaml_config_file(
+            path.join(path.dirname(comp_file), 'services.yaml'))
+        DESCRIPTION_CACHE[comp_file] = descriptions
+
+    if domain == ha.DOMAIN:
+        description = descriptions[domain].get(service, {})
+    else:
+        description = descriptions.get(service, {})
+
+    return {
+        'description': description.get('description', ''),
+        'fields': description.get('fields', {})
+    }
+
+
+@asyncio.coroutine
+@bind_hass
+def async_get_all_descriptions(hass):
+    """Return the descriptions for all service calls."""
+    services = hass.services.async_services()
+    descriptions = {}
+    for domain in services:
+        descriptions[domain] = {}
+        for service in services[domain]:
+            description = yield from hass.async_add_job(
+                get_description, hass, domain, service)
```

Comment:
Since we know that we're going to load all the descriptions, it would help if we preload the catch all services file (`/components/services.yaml`). That file is pretty heavy if we have to parse it X times.

Reply:
It should only be read and parsed once, that's what the `DESCRIPTION_CACHE` is for.

Reply:
Oh I see now that you cache that file. I'm not sure if that is a great idea, it's big and a lot will be unused.